cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -423,6 +435,8 @@ private File saveCConf(CConfiguration conf, File file) throws IOException {
     // Set the CFG_LOCAL_DATA_DIR to a relative path as the data directory for the container should be relative to the
     // container directory
     copied.set(Constants.CFG_LOCAL_DATA_DIR, "data");
+
+    copied.setStrings(Constants.AppFabric.PROGRAM_CONTAINER_DIST_JARS, (String[]) extraJars.toArray());
```

Comment:
Call `extraJars.toArray(new String[extraJars.size()])` instead without the need for casting.